### PR TITLE
Moves can_synth to chemical_flags

### PR DESCRIPTION
**Pull request by spessbro**
Changes the can_synth var to be a bitflag in chemical_flags remember to
check for initial()

## About The Pull Request
Refactors the can_synth var in reagents to chemical_flags and changes the default to false so we don't get drug and reagent and other weird chems like that

## Why It's Good For The Game
oranges said good :)

## Changelog
:cl:
refactor: refactored can_synth into chemical_flags
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/impure_reagents.dm [hunk 1]
```diff
@@ -23,6 +23,7 @@
 	name = "Toxic sludge"
 	description = "Toxic chemical isomers made from impure reactions. Causes toxin damage"
 	ph = 2
+	chemical_flags = REAGENT_CANSYNTH
```
Comment (Thalpy) [inline code/modules/reagents/chemistry/reagents/impure_reagents.dm:26]: Will this override the parent's flags?
i.e. all children of impurity should have `chemical_flags = REAGENT_INVISIBLE | REAGENT_SNEAKYNAME | REAGENT_DONOTSPLIT` as well

Comment (spessbro) [reply code/modules/reagents/chemistry/reagents/impure_reagents.dm]: my brain smal

Comment (Thalpy) [reply code/modules/reagents/chemistry/reagents/impure_reagents.dm]: Your brain strong!

**Comment by necromanceranne** [comment]
Since there is an absurd amount of reagents, wouldn't it make more sense to make the flag for what chems _can't_ be synthed?

**Comment by XDTM** [comment]
considering the ones that shouldn't be synthed are the parents of all the others, i imagine it would be tricky unless they're handled by a separate blacklist

**Comment by Jared-Fogle** [comment]
@necromanceranne 

> Since there is an absurd amount of reagents, wouldn't it make more sense to make the flag for what chems can't be synthed?

No, one of the motivations behind this is it makes it explicit that you want a chemical to be able to be synthesized. It is now impossible to do incidentally.

**Comment by necromanceranne** [comment]
> considering the ones that shouldn't be synthed are the parents of all the others, i imagine it would be tricky unless they're handled by a separate blacklist

Oh right I forgot how our reagents inheritance is like. Fair enough.

File: code/_globalvars/bitfields.dm [hunk 2]
```diff
@@ -301,6 +301,7 @@ DEFINE_BITFIELD(chemical_flags, list(
 	"REAGENT_INVISIBLE" = REAGENT_INVISIBLE,
 	"REAGENT_SNEAKYNAME" = REAGENT_SNEAKYNAME,
 	"REAGENT_SPLITRETAINVOL" = REAGENT_SPLITRETAINVOL,
+	"REAGENT_CANSYNTH" = REAGENT_CANSYNTH,
```
Comment (Rohesie) [inline code/_globalvars/bitfields.dm:304]:
```suggestion
	"REAGENT_CAN_BE_SYNTHESIZED" = REAGENT_CAN_BE_SYNTHESIZED,
```
Clear names, please.

**Review by Rohesie — CHANGES_REQUESTED** [review]
Just a small mass-replacement.

**Comment by spessbro** [comment]
no

**Comment by Thalpy** [comment]
What

**Comment by Rohesie** [comment]
> 
> 
> What

Their git-fu is weak. They had to remake the PR:
https://github.com/tgstation/tgstation/pull/56891

**Comment by spessbro** [comment]
> Their git-fu is weak. They had to remake the PR:

how dare you out me like this

**Pull request closed without merging**